cranelift/filetests/filetests/isa/x64/simd-lane-access-compile.clif

```diff
@@ -252,7 +252,7 @@ block0(v0: f64):
 ;   pushq   %rbp
 ;   movq    %rsp, %rbp
 ; block0:
-;   movddup %xmm0, %xmm0
+;   pshufd  $68, %xmm0, %xmm0
```

Comment:
Something feels off here: I look at the top of the file and see `has_ssse3` but then here `movddup` is not used. Is it that the rule ordering should be reversed? I.e., the `movddup` rule should be higher priority than the `pshufd` rule?

Reply:
Ok, looking into this more I think this is the right change: we use `PSHUFD` for the XMM-to-XMM splats (slightly higher throughput) and `MOVDDUP` for the MEM-to-XMM splats (slightly lower latency and higher throughput).